packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/__tests__/MappingTest.test.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  ObjectInputData,
+  OBJECT_INPUT_TYPE,
+} from '../../../../../models/metamodels/pure/model/packageableElements/store/modelToModel/mapping/ObjectInputData';
+import { getRawGraphFetchTreeData } from '../../../../shared/RawGraphFetchTreeUtil';
+import { getTestEditorStore } from '../../../../StoreTestUtils';
+import { MappingEditorState } from '../MappingEditorState';
+import {
+  MappingExecutionGraphFetchQueryState,
+  MappingExecutionJsonModelConnectionRuntimeState,
+} from '../MappingExecutionState';
+import {
+  mappingTestData,
+  executeResultResponse,
+  generatedSourceData,
+} from './MappingTestData';
+
+test(
+  unitTest(
+    'Test creating mapping test and saving/displaying input and expected result data',
+  ),
+  async () => {
+    // setup up test and open mapping editor
+    const editorStore = getTestEditorStore();
+    await editorStore.graphState.initializeSystem();
+    await editorStore.graphState.graphManager.buildGraph(
+      editorStore.graphState.graph,
+      mappingTestData,
+      { TEMPORARY__keepSectionIndex: true },
+    );
+    const result = losslessParse(executeResultResponse) as JsonExecutionResult;
+    const jsonResult = new JsonExecutionResult(result.values);
+    jest
+      .spyOn(editorStore.graphState.graphManager, 'executeMapping')
+      .mockResolvedValue(jsonResult);
+    jest
+      .spyOn(editorStore.graphState.graphManager, 'getEngineConfig')
+      .mockReturnValue(new AbstractEngineConfig());
+    const mapping = editorStore.graphState.graph.getMapping(
+      'model::SimpleMapping',
+    );
+    const targetClass = editorStore.graphState.graph.getClass(
+      'model::Target_Person',
+    );
+    const sourceClass = editorStore.graphState.graph.getClass(
+      'model::Source_Person',
+    );
+    editorStore.openElement(mapping);
+    const mappingEditorState = editorStore.getCurrentEditorState(
+      MappingEditorState,
+    );
+    const executionState = mappingEditorState.executionState;
+    // set target class and build graph fetch tree
+    const queryState = new MappingExecutionGraphFetchQueryState(editorStore);
+    queryState.setTarget(targetClass);
+    queryState.setGraphFetchTree(
+      getRawGraphFetchTreeData(editorStore, targetClass, mapping),
+    );
+    // populate test data
+    const runtimeState = new MappingExecutionJsonModelConnectionRuntimeState(
+      editorStore,
+      mapping,
+    );
+    runtimeState.setSourceClass(sourceClass);
+    runtimeState.setTestData(JSON.stringify(generatedSourceData, undefined, 2));
+    executionState.setRuntimeState(runtimeState);
+    const testData = JSON.parse(runtimeState.testData) as Record<
+      string,
+      string
+    >;
+    expect(testData.sFirstName).toEqual('myFirstName');
+    expect(testData.sLastName).toEqual('myLastName');
+    executionState.setQueryState(queryState);
+    const expectedTestDataDisplay =
```

Comment:
why don't we take this chance to test for the decimal rounding?

Reply:
I believe we have separate tests for the decimal rounding when the feature was first added